aten/src/ATen/core/rref_interface.h

```diff
@@ -1,14 +1,16 @@
 #pragma once
 
-#include <torch/csrc/distributed/rpc/types.h>
+#include <c10/util/intrusive_ptr.h>
 
-namespace torch {
-namespace distributed {
-namespace rpc {
+namespace c10 {
+
+struct Type;
+using TypePtr = std::shared_ptr<Type>;
+using worker_id_t = int16_t;
 
 // This abstract class contains only user-facing APIs, and will be shared
 // between jit and distributed to implement TorchScript support.
-class RRefInterface {
+class C10_EXPORT RRefInterface : public c10::intrusive_ptr_target {
```

Comment:
what's the implication of inheriting from intrusive_ptr_target? Does this mean `RRefInterface` will always be a shared_ptr like object?